activerecord/lib/active_record/railties/databases.rake

```diff
@@ -449,16 +449,15 @@ db_namespace = namespace :db do
     task dump: :load_config do
       ActiveRecord::Tasks::DatabaseTasks.with_temporary_pool_for_each do |pool|
         db_config = pool.db_config
-        schema_format = ENV.fetch("SCHEMA_FORMAT", ActiveRecord.schema_format).to_sym
-        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config, schema_format)
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config, ENV["SCHEMA_FORMAT"] || db_config.schema_format)
       end
 
       db_namespace["schema:dump"].reenable
     end
 
     desc "Load a database schema file (either db/schema.rb or db/structure.sql, depending on `ENV['SCHEMA_FORMAT']` or `config.active_record.schema_format`) into the database"
     task load: [:load_config, :check_protected_environments] do
-      ActiveRecord::Tasks::DatabaseTasks.load_schema_current(ActiveRecord.schema_format, ENV["SCHEMA"])
+      ActiveRecord::Tasks::DatabaseTasks.load_schema_current(ENV["SCHEMA"])
```

Comment:
Here you're passing the file as the first positional argument, which is format. I think you need to pass a nil as the first arg for now, and file as the second arg.